src/core/server/http/http_server.ts

```diff
@@ -190,6 +190,20 @@ export class HttpServer {
     this.server = undefined;
   }
 
+  private getAuthOption(
+    authRequired: RouteConfigOptions<any>['authRequired'] = true
+  ): false | { mode: 'required' | 'optional' } {
+    if (this.authRegistered) {
```

Comment:
It sets mode explicitly. I had to change the logic to inline it with `mode: optional` declaration since it cannot be used without auth strategy registration. the previous implementation relies on the internal hapi logic that `true === undefined`.